src/OpenTelemetry.Exporter.Geneva/GenevaLogExporter.cs

```diff
@@ -443,5 +489,66 @@ private static byte GetSeverityNumber(LogLevel logLevel)
                 return 1;
         }
     }
+
+    private static int Sanitize(byte[] buffer, int cursor, ref int cursorStartIdx, ref int validNameLength, string categoryName)
+    {
+        if (categoryName.Length > (1 << 8) - 1)
+        {
+            // The size of categoryName should not be greater than 2^8 -1.
+            return cursor;
+        }
+
+        // Reserve 2 bytes for storing LIMIT_MAX_STR8_LENGTH_IN_BYTES and (byte)validNameLength -
+        // these 2 bytes will be back filled after iterating through categoryName.
+        cursor += 2;
+
+        // Special treatment for the first character.
+        var firstChar = categoryName[0];
+        if (firstChar >= 'A' && firstChar <= 'Z')
+        {
+            buffer[cursor++] = (byte)firstChar;
+            ++validNameLength;
+        }
+        else if (firstChar >= 'a' && firstChar <= 'z')
+        {
+            // If the first character in the resulting string is lower-case ALPHA,
+            // it will be converted to the corresponding upper-case.
+            buffer[cursor++] = (byte)(firstChar - 32);
+            ++validNameLength;
+        }
+        else
+        {
+            // Not a valid name - Part B name should follow PascalCase naming convention.
+            return cursor -= 2;
+        }
+
+        const int maxNameCharCount = 50;
+        for (int i = 1; i < categoryName.Length; ++i)
+        {
+            if (validNameLength >= maxNameCharCount - 1)
+            {
+                break;
+            }
+
+            var cur = categoryName[i];
+            if ((cur >= '0' && cur <= '9') || (cur >= 'A' && cur <= 'Z') || (cur >= 'a' && cur <= 'z'))
+            {
+                buffer[cursor++] = (byte)cur;
+                ++validNameLength;
+            }
+        }
+
+        if (validNameLength > 0)
+        {
+            // Backfilling MessagePack serialization protocol and valid category length to the startIdx of the cateoryName byte array.
+            MessagePackSerializer.BackFill(buffer, cursorStartIdx, validNameLength);
+        }
+        else
+        {
+            cursor -= 2;
```

Comment:
Would this line be hit? You're returning on `validNameLength == 0` on line 522 already.

Reply:
Good catch! Thank you.